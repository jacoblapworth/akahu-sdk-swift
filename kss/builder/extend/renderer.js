const { regexDictionary, types } = require('../parser/constants');
const tinycolor = require('tinycolor2');

module.exports = function(handlebars) {
	/**
	 * Extracts the variable properties from the line
	 * returns:
	 *  token name, token value, and optional [ token example ]
	 */
	handlebars.registerHelper('renderTokens', renderTokens);

	/**
	 * Compares the name to see if the container needs to be rendered as inverted.
	 * Similar to the isEqual
	 *
	 * e.g.
	 * {{#isInverted name}}
	 * 		ADD THIS
	 * {{else}} [optional]
	 * 		WILL EXECUTE THIS
	 * {{/isInverted}}
	 */
	handlebars.registerHelper('isInverted', isInverted);
};

const testString = 'To understand what recursion is, you must first understand recursion.';


function renderTokens(variables, block) {
	var newBlock = '';
	variables.split('\n').forEach(line => {
		var test;
		const matches = line.match(/[^:]+/g).map(match => match.trim());

		if (matches[1]) {
			this.token = {};
			this.token.name = matches[0];
			this.token.value = matches[1];
			this.token.title = matches[3];
			this.token.description = matches[4];
			if (matches[2] != undefined) {
				this.token.example = getExample(matches[2], this.token);
			}
			newBlock += block.fn(this);
		}
	});
	return newBlock;
}

function getExample(type, token) {
	switch(type) {
		case types.color:
			const codeTextClass = tinycolor(token.value).isDark() ?
				'ds-colour-code-light'
				: null;
			return `
					<div class="ds-colour-example" style="background-color:${token.value}">
						<code class="${codeTextClass}">${token.value}</code>
						<code class="${codeTextClass}">${token.name}</code>
					</div>`;
		case types.border:
			return `
				<div class="ds-colour-example" style="border:${token.value}">
					<code>${token.name}</code>
				</div>`;
		case types.shadow:
			return `
					<div class="ds-colour-example" style="box-shadow:${token.value}">
						<code>${token.name}</code>
					</div>`;
		case types.fontSize:
			return  `<span class="xui-text-truncated" style="font-size:${token.value};">Aa</span>`;
		case types.fontWeight:
			return `<span style="font-weight:${token.value};">Aa</span>`;
		case types.lineHeight:
			return `<span style="line-height:${token.value}">${testString}</span>`;
		case types.spacing:
			return `<div class="ds-spacing-example" style="height:${token.value};width:${token.value};"></div>`;
		default:
			return '';
	}
}

function isInverted(name, options) {
	return regexDictionary.inverted.test(name) ? options.fn(this) : options.inverse(this);
}
