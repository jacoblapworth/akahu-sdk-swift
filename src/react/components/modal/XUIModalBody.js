import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { baseClass } from './constants';

const XUIModalBody = ({ className, children, qaHook }) => {
  const classNames = cn(`${baseClass}--body`, className);

  return (
    <div className={classNames} data-automationid={qaHook}>
      {children}
    </div>
  );
};

export default XUIModalBody;

XUIModalBody.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  qaHook: PropTypes.string,
};
