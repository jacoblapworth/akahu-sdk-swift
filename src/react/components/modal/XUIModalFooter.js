import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

export default function XUIModalFooter({ className, children, defaultLayout }) {
	const classNames = cn(
		'xui-modal--footer',
		className,
		{ ['xui-padding-large'] : defaultLayout }
	);

	return (
		<footer className={classNames}>
			{children}
		</footer>
	);
}

XUIModalFooter.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,

	/** If the modal will use the default XUI style layout */
	defaultLayout: PropTypes.bool
};

XUIModalFooter.defaultProps = {
	defaultLayout: true,
};
