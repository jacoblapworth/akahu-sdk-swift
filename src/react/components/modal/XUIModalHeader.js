import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { baseClass } from './constants';
import { ns } from '../helpers/xuiClassNamespace';

export default function XUIModalHeader({
	className,
	children,
	defaultLayout,
	qaHook,
	id,
}) {
	const classNames = cn(
		`${baseClass}--header`,
		`${baseClass}--heading`,
		defaultLayout && `${ns}-padding-left-large`,
		className,
	);
	return (
		<header id={id} className={classNames} data-automationid={qaHook}>
			{children}
		</header>
	);
}

XUIModalHeader.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	qaHook: PropTypes.string,

	/** Id for the modal header. Used for automatically providing a label to the modal. */
	id: PropTypes.string,

	/** If the modal will use the default XUI style layout */
	defaultLayout: PropTypes.bool,
};

XUIModalHeader.defaultProps = {
	defaultLayout: true,
};
