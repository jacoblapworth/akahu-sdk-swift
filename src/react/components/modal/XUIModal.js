import React, { Component, Children, cloneElement } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import verge from 'verge';
import { Portal } from 'react-portal';
import cross from '@xero/xui-icon/icons/cross';
import XUIIcon from '../icon/XUIIcon';
import XUIButton from '../button/XUIButton';
import XUIModalHeader from './XUIModalHeader';
import { lockScroll, unlockScroll } from '../helpers/lockScroll';
import portalContainer, { portalClass } from '../helpers/portalContainer';
import {baseClass} from './constants';
import {ns} from "../helpers/xuiClassNamespace";

export const modalSizes = {
	default: `${baseClass}-width-default`,
	small: `${baseClass}-width-small`,
	medium: `${baseClass}-width-medium`,
	large: `${baseClass}-width-large`,
	xlarge: `${baseClass}-width-xlarge`,
	fullscreen: `${baseClass}-fullscreen`
};

/**
 * Predicate to determine if the props of a modal have changed in such as way as to necessitate
 * adding/removing global event listeners.
 *
 * @private
 * @param {Object} props
 * @param {Object} otherProps
 * @returns {Boolean}
 */
function shouldUpdateListeners(props, otherProps) {
	return (
		!!props.onClose !== !!otherProps.onClose ||
		props.hideOnEsc !== otherProps.hideOnEsc ||
		props.restrictFocus !== otherProps.restrictFocus ||
		props.keyListenerTarget !== otherProps.keyListenerTarget
	);
}

/**
 * Add the global event listeners necessary to ensure that props like hideOnEsc
 * can function properly.
 *
 * @private
 * @param {XUIModal} modal
 */
function addListeners(modal) {
	const { onClose, keyListenerTarget, restrictFocus, hideOnEsc } = modal.props;
	const listenerTarget = keyListenerTarget == null ? window : keyListenerTarget;

	// Be paranoid.  Some test environments won't have a window object.
	if (listenerTarget != null) {
		if (onClose && hideOnEsc) {
			modal._keyUpHandler = modal._keyUpHandler.bind(modal);
			listenerTarget.addEventListener('keyup', modal._keyUpHandler);
		}
		if (restrictFocus) {
			modal._restrictFocus = modal._restrictFocus.bind(modal);
			listenerTarget.addEventListener('focus', modal._restrictFocus, true);
		}
	}
}

/**
 * Remove any global event listeners associated with a given modal.
 *
 * @private
 * @param {XUIModal} modal
 */
function removeListeners(modal) {
	const { keyListenerTarget } = modal.props;
	const listenerTarget = keyListenerTarget == null ? window : keyListenerTarget;

	// Be paranoid.  Some test environments won't have a window object.
	if (listenerTarget != null) {
		listenerTarget.removeEventListener('keyup', modal._keyUpHandler);
		listenerTarget.removeEventListener('focus', modal._restrictFocus, true);
	}
}

export default class XUIModal extends Component {

	state = { positionSettings: null }

	componentDidMount() {
		const modal = this;
		addListeners(this);
		if (modal.props.isOpen) {
			lockScroll();
			modal._isScrollLocked = true;

			const calcOffsetTop = (modal) => {
				const viewportH = verge.viewportH();
				const modalHeight = modal._modalNode.getBoundingClientRect().height;
				return Math.max(((viewportH - modalHeight) / 2) - 15, 0); // subtracts 15px ($xui-s-standard) from `top` to take into account XUIMask's (wrapping component) existing padding
			}
			const positionSettings = {
				top: `${calcOffsetTop(modal)}px`,
				marginTop: 0
			}

			modal.setState({
				positionSettings
			});

			if (!modal._maskNode.contains(document.activeElement)) {
				modal._modalNode.focus();
			}
		}
	}

	componentWillUnmount() {
		removeListeners(this);

		unlockScroll();
		this._isScrollLocked = false;
	}

	componentWillReceiveProps(nextProps) {
		if (shouldUpdateListeners(this.props, nextProps)) {
			removeListeners(this);
		}

	}


	componentDidUpdate(prevProps) {
		const modal = this;
		const { isOpen, restrictFocus } = modal.props;

		if (shouldUpdateListeners(this.props, prevProps)) {
			addListeners(this);
		}

		if (isOpen && !modal._isScrollLocked) {
			lockScroll();
			modal._isScrollLocked = true;
		}

		if (!isOpen && modal._isScrollLocked) {
			unlockScroll();
			modal._isScrollLocked = false;
		}

		if (
			isOpen &&
			restrictFocus &&
			(!prevProps.isOpen || !prevProps.restrictFocus) &&
			!modal._maskNode.contains(document.activeElement)
		) {
			modal._modalNode.focus();
		}
	}

	/**
	 * @private
	 * @param {Object} event - A keyUp event which is set to be listened to by componentDidMount.
	 * If the modal needs to close when the user presses the esc key, we need to attach a key
	 * listener to a parent (`window` by default) to catch this key press
	 */
	_keyUpHandler(event) {
		const { isOpen, onClose } = this.props;
		const escapeKeyCode = 27;
		if (event.keyCode === escapeKeyCode && isOpen && onClose) {
			onClose();
		}
	}

	/**
	 * @private
	 * @param {Object} event - A focus change event which is set to be listened to by componentDidMount
	 * Limits the focusable elements to those within the modal
	 **/
	_restrictFocus(event) {
		const modal = this;
		const { isOpen, restrictFocus } = modal.props;
		if (isOpen && restrictFocus) {
			const maskNode = document.querySelector(`.${portalClass}`);
			const targetIsWindow = event.target === window;
			if (targetIsWindow || !maskNode.contains(event.target)) {
				event.stopPropagation();
				if (modal._modalNode) {
					modal._modalNode.focus();
				}
			}
		}
	}

	render() {
		const {
			hideOnOverlayClick,
			onClose,
			size,
			isOpen,
			className,
			maskClassName,
			closeClassName,
			children,
			defaultLayout,
			ariaLabelledBy,
			ariaDescribedBy,
			qaHook,
			id,
			isForm,
			isUsingPortal
		} = this.props;
		const {
			positionSettings
		} = this.state;

		const maskClasses = cn(
			`${ns}-mask`,
			maskClassName,
			isOpen && `${ns}-mask-is-active`
		);
		const modalClasses = cn(
			baseClass,
			modalSizes[size],
			defaultLayout && `${baseClass}-layout`,
			className
		);
		const overlayClickHandler =
			hideOnOverlayClick && onClose
				? function(event) {
						if (event.target.classList.contains(`${ns}-mask`) && isOpen) {
							onClose();
						}
					}
				: null;

		const closeButton = onClose ? (
			<XUIButton
				qaHook={qaHook && `${qaHook}--close`}
				onClick={onClose}
				title="Close"
				className={cn(`${baseClass}--close`, closeClassName)}
				key="close-button"
				type="button"
				variant="icon"
			>
				<XUIIcon icon={cross} isBoxed />
			</XUIButton>
		) : null;
		let containsHeader = false;
		const finalChildren = Children.map(children, child => {
			if (child && child.type === XUIModalHeader) {
				containsHeader = true;
				return cloneElement(child, { ...child.props }, [
					child.props.children,
					closeButton
				]);
			} else {
				return child;
			}
		});
		const MainElement = isForm ? 'form' : 'section';
		const childNodes = (
			<div
				id={id}
				className={maskClasses}
				onClick={overlayClickHandler}
				aria-hidden={!isOpen}
				data-automationid={qaHook && `${qaHook}--mask`}
				ref={m => (this._maskNode = m)}
			>
				<MainElement
					className={modalClasses}
					tabIndex={isOpen ? 0 : -1}
					role={isOpen ? 'dialog' : null}
					style={positionSettings}
					aria-labelledby={ariaLabelledBy}
					aria-describedby={ariaDescribedBy}
					data-automationid={qaHook}
					ref={m => (this._modalNode = m)}
				>
					{!containsHeader ? (
						<XUIModalHeader qaHook={qaHook && `${qaHook}--header`}>
							{closeButton}
						</XUIModalHeader>
					) : null}
					{finalChildren}
				</MainElement>
			</div>
		);

		return isUsingPortal ? (
			isOpen ? (
				<Portal node={portalContainer()}>
					<div
						className={`${ns}-container`}
						data-automationid={qaHook && `${qaHook}--container`}
					>
						{childNodes}
					</div>
				</Portal>
			) : null
		) : (
			childNodes
		);
	}
}

XUIModal.propTypes = {
	/** If the modal will be hidden when the user presses the Esc key */
	hideOnEsc: PropTypes.bool,

	/** If the modal will be hidden when the user clicks the overlay */
	hideOnOverlayClick: PropTypes.bool,

	/** Bind a function to fire when the modal requests to be hidden */
	onClose: PropTypes.func,

	/** The size (aka width) of this modal */
	size: PropTypes.oneOf(['default', 'small', 'medium', 'large', 'xlarge', 'fullscreen']),

	/** Whether the modal is visible */
	isOpen: PropTypes.bool,

	/** Whether the modal wrapping element should be a `<form>` rather than a `<section>`. Allows the enter key to activate the submit button inside native form controls. */
	isForm: PropTypes.bool,

	/** The target that should listen to key presses. Defaults to the window. */
	keyListenerTarget: PropTypes.object,

	/** Custom classes for the mask */
	maskClassName: PropTypes.string,

	/** Custom classes for the close button */
	closeClassName: PropTypes.string,

	/** Restricts focus to elements within the modal while it is open */
	restrictFocus: PropTypes.bool,

	/** ID for the element containing an appropriate label for screen readers */
	ariaLabelledBy: PropTypes.string,

	/** ID for the element containing an appropriate description for screen readers */
	ariaDescribedBy: PropTypes.string,

	/** Renders the modal to the bottom of the current document when true. Otherwise inline. */
	isUsingPortal: PropTypes.bool,

	/** If the modal will use the default XUI style layout */
	defaultLayout: PropTypes.bool,

	children: PropTypes.node,
	className: PropTypes.string,
	id: PropTypes.string,
	qaHook: PropTypes.string
};

XUIModal.defaultProps = {
	hideOnEsc: true,
	hideOnOverlayClick: false,
	isOpen: false,
	size: 'medium',
	defaultLayout: true,
	restrictFocus: true,
	isUsingPortal: true,
	isForm: false
};
