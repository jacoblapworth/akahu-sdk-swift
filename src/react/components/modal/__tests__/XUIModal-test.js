import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import XUIModal from '../XUIModal';
import XUIModalBody from '../XUIModalBody';
import XUIModalFooter from '../XUIModalFooter';
import XUIModalHeader from '../XUIModalHeader';
import uuidv4 from 'uuid/v4';

jest.mock('uuid/v4');
uuidv4.mockImplementation(() => 'generatedHeaderId');

Enzyme.configure({ adapter: new Adapter() });

const NOOP = () => {};

// NOTE: The esc key functionality has not been tested here as event listeners don't work very well in React testing
describe('XUIModal', () => {
  it('Should be possible to do snapshot testing on the modal', () => {
    const component = renderer.create(
      <XUIModal isUsingPortal={false} id="test-modal" qaHook="xui-modal" onClose={NOOP}>
        <XUIModalHeader>Test header</XUIModalHeader>
        <div>test</div>
      </XUIModal>,
    );
    expect(component).toMatchSnapshot();
  });

  it('Should render with any additional classes provided through the className, maskClassName and closeClassName props', function() {
    const component = mount(
      <XUIModal
        isUsingPortal={false}
        id="test-modal"
        className="classyMcClassFace"
        maskClassName="maskClassyMcClassFace"
        onClose={NOOP}
        closeClassName="closeClassyMcClassFace"
      />,
    );

    const domNode = component.find('.xui-modal');
    const maskNode = component.find('.xui-mask');

    expect(domNode.hasClass('classyMcClassFace')).toBeTruthy();
    expect(maskNode.hasClass('maskClassyMcClassFace')).toBeTruthy();
    expect(domNode.find('button').hasClass('closeClassyMcClassFace')).toBeTruthy();
  });

  it('Should render with the appropriate size modifier', function() {
    const component = mount(
      <div>
        <XUIModal isUsingPortal={false} id="test-modal-small" size="small" onClose={NOOP} />
        <XUIModal isUsingPortal={false} id="test-modal-medium" size="medium" onClose={NOOP} />
        <XUIModal isUsingPortal={false} id="test-modal-large" size="large" onClose={NOOP} />
        <XUIModal isUsingPortal={false} id="test-modal-xlarge" size="xlarge" onClose={NOOP} />
        <XUIModal
          isUsingPortal={false}
          id="test-modal-fullscreen"
          size="fullscreen"
          onClose={NOOP}
        />
      </div>,
    );

    const modal1 = component.find('#test-modal-small .xui-modal');
    const modal2 = component.find('#test-modal-medium .xui-modal');
    const modal3 = component.find('#test-modal-large .xui-modal');
    const modalXLarge = component.find('#test-modal-xlarge .xui-modal');
    const modalFullScreen = component.find('#test-modal-fullscreen .xui-modal');

    expect(modal1.hasClass('xui-modal-width-small')).toBeTruthy();
    expect(modal1.hasClass('xui-modal-width-medium')).toBeFalsy();
    expect(modal1.hasClass('xui-modal-width-large')).toBeFalsy();
    expect(modal1.hasClass('xui-modal-width-xlarge')).toBeFalsy();
    expect(modal1.hasClass('xui-modal-fullscreen')).toBeFalsy();

    expect(modal2.hasClass('xui-modal-width-small')).toBeFalsy();
    expect(modal2.hasClass('xui-modal-width-medium')).toBeTruthy();
    expect(modal2.hasClass('xui-modal-width-large')).toBeFalsy();
    expect(modal2.hasClass('xui-modal-width-xlarge')).toBeFalsy();
    expect(modal2.hasClass('xui-modal-fullscreen')).toBeFalsy();

    expect(modal3.hasClass('xui-modal-width-small')).toBeFalsy();
    expect(modal3.hasClass('xui-modal-width-medium')).toBeFalsy();
    expect(modal3.hasClass('xui-modal-width-large')).toBeTruthy();
    expect(modal3.hasClass('xui-modal-width-xlarge')).toBeFalsy();
    expect(modal3.hasClass('xui-modal-fullscreen')).toBeFalsy();

    expect(modalXLarge.hasClass('xui-modal-width-small')).toBeFalsy();
    expect(modalXLarge.hasClass('xui-modal-width-medium')).toBeFalsy();
    expect(modalXLarge.hasClass('xui-modal-width-large')).toBeFalsy();
    expect(modalXLarge.hasClass('xui-modal-width-xlarge')).toBeTruthy();
    expect(modalXLarge.hasClass('xui-modal-fullscreen')).toBeFalsy();

    expect(modalFullScreen.hasClass('xui-modal-width-small')).toBeFalsy();
    expect(modalFullScreen.hasClass('xui-modal-width-medium')).toBeFalsy();
    expect(modalFullScreen.hasClass('xui-modal-width-large')).toBeFalsy();
    expect(modalFullScreen.hasClass('xui-modal-width-xlarge')).toBeFalsy();
    expect(modalFullScreen.hasClass('xui-modal-fullscreen')).toBeTruthy();
  });

  it('Should correctly pass close handlers to the close button and mask', function() {
    let clicked = false;
    const demoFunction = () => {
      clicked = true;
    };

    const component = mount(
      <XUIModal
        isUsingPortal={false}
        onClose={demoFunction}
        isOpen={true}
        hideOnOverlayClick={true}
      />,
    );
    const maskNode = component.find('.xui-mask');
    maskNode.simulate('click');
    expect(clicked).toEqual(true);

    clicked = false;

    const closeButtonNode = maskNode.find('.xui-button');
    closeButtonNode.simulate('click');
    expect(clicked).toEqual(true);
  });

  it('Should not fire onClose on mask click if hideOnOverlayClick is set to false', function() {
    let maskClicked = false;
    const demoFunction = () => {
      maskClicked = true;
    };

    const component = mount(
      <XUIModal
        isUsingPortal={false}
        id="test-modal-overlayClick"
        onClose={demoFunction}
        hideOnOverlayClick={false}
        isOpen={true}
      />,
    );
    const maskNode = component.find('.xui-mask');
    maskNode.simulate('click');
    setTimeout(() => {
      expect(maskClicked).toBeFalsy();
    }, 0);
  });

  it('Should fire on close on click of the mask by default', () => {
    let maskClicked = false;
    const demoFunction = () => {
      maskClicked = true;
    };

    const component = mount(
      <XUIModal
        isUsingPortal={false}
        id="test-modal-overlayClick"
        onClose={demoFunction}
        isOpen={true}
      />,
    );
    const maskNode = component.find('.xui-mask');
    maskNode.simulate('click');
    setTimeout(() => {
      expect(maskClicked).toBeTruthy();
    }, 0);
  });

  it('Should only add default layout classes if defaultLayout is explicitly set to true or by default', function() {
    const component = mount(
      <div>
        <XUIModal
          isUsingPortal={false}
          id="test-modal-nolayout"
          defaultLayout={false}
          isOpen={true}
          onClose={NOOP}
        />
        <XUIModal isUsingPortal={false} id="test-modal-layout" onClose={NOOP} />
      </div>,
    );
    const firstModal = component.find('#test-modal-nolayout .xui-modal');
    const secondModal = component.find('#test-modal-layout .xui-modal');

    expect(firstModal.hasClass('xui-modal-layout')).toBeFalsy();
    expect(secondModal.hasClass('xui-modal-layout')).toBeTruthy();
  });

  it('Should not add margin or padding classes to subcomponents if defaultLayout is set to false', function() {
    const component = mount(
      <XUIModal
        isUsingPortal={false}
        defaultLayout={false}
        id="test-modal-layoutSecond"
        isOpen={true}
      >
        <XUIModalHeader defaultLayout={false} />
        <XUIModalBody defaultLayout={false} />
        <XUIModalFooter defaultLayout={false} />
      </XUIModal>,
    );
    const modal = component.find('#test-modal-layoutSecond .xui-modal');

    for (let i = 0; i < modal.children.length; i++) {
      expect(modal.childAt(i).hasClass('xui-modal-layout')).toBeFalsy();
    }
  });

  it('Should render as a form when isForm is true', function() {
    const component = mount(
      <XUIModal isUsingPortal={false} isForm={true} id="test-modal" isOpen={true}>
        <XUIModalHeader />
        <XUIModalBody />
        <XUIModalFooter />
      </XUIModal>,
    );

    const domNode = component.find('.xui-modal');
    expect(domNode.is('form')).toBeTruthy();
  });

  it('should not render as a form by default, or if isForm is set to false', () => {
    const component = mount(
      <XUIModal isUsingPortal={false} id="test-modal">
        <XUIModalHeader />
        <XUIModalBody />
        <XUIModalFooter />
      </XUIModal>,
    );

    const domNode = component.find('.xui-modal');
    expect(domNode.is('form')).toBeFalsy();

    const componentSet = mount(
      <XUIModal isUsingPortal={false} isForm={false} id="test-modal">
        <XUIModalHeader />
        <XUIModalBody />
        <XUIModalFooter />
      </XUIModal>,
    );

    const secondDomNode = componentSet.find('.xui-modal');
    expect(secondDomNode.is('form')).toBeFalsy();
  });

  it('should render an automation id when a qaHook is passed in', () => {
    const automationid = renderer.create(
      <XUIModal isUsingPortal={false} qaHook="xui-modal" onClose={NOOP}>
        <div>test</div>
      </XUIModal>,
    );

    expect(automationid).toMatchSnapshot();
  });

  it('should lock the scroll of a window when mounted and unlocked when unmounted', () => {
    const modalMounted = mount(
      <XUIModal isUsingPortal={false} onClose={NOOP} isOpen={true}>
        <div>test</div>
      </XUIModal>,
    );

    expect(modalMounted.instance()._isScrollLocked).toBeTruthy();

    modalMounted.setProps({ isOpen: false });

    expect(modalMounted.instance()._isScrollLocked).toBeFalsy();
  });

  it('should render closed by default', () => {
    const modalMounted = shallow(
      <XUIModal isUsingPortal={false} onClose={NOOP}>
        <div>test</div>
      </XUIModal>,
    );

    expect(modalMounted.hasClass('xui-mask-is-active')).toBeFalsy();
  });

  it('should render open when isOpen is set to true', () => {
    const modalMounted = mount(
      <XUIModal isUsingPortal={false} onClose={NOOP} isOpen={true}>
        <div>test</div>
      </XUIModal>,
    );

    expect(modalMounted.find('.xui-mask').hasClass('xui-mask-is-active')).toBeTruthy();
  });
});
