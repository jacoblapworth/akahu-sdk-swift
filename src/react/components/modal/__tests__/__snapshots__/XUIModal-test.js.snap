// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XUIModal Should be possible to do snapshot testing on the modal 1`] = `
<div
  aria-hidden={true}
  className="xui-mask xui-unmask"
  data-automationid="xui-modal--mask"
  id="test-modal"
  onClick={null}
  role="presentation"
>
  <section
    aria-labelledby="generatedHeaderId"
    className="xui-modal xui-modal-width-medium xui-modal-layout"
    data-automationid="xui-modal"
    role={null}
    style={null}
    tabIndex={-1}
  >
    <header
      className="xui-modal--header xui-modal--heading"
      id="generatedHeaderId"
    >
      Test header
      <button
        className="xui-button xui-modal--close xui-button-icon xui-button-icon-medium"
        data-automationid="xui-modal--close"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <div
          className="xui-iconwrapper xui-iconwrapper-medium"
        >
          <svg
            className="xui-icon"
            focusable="false"
            height={12}
            viewBox="0 0 12 12"
            width={12}
          >
            <path
              d="M6 4.5L10.5 0 12 1.5 7.5 6l4.5 4.5-1.5 1.5L6 7.5 1.5 12 0 10.5 4.5 6 0 1.5 1.5 0 6 4.5z"
              role="presentation"
            />
          </svg>
        </div>
      </button>
    </header>
    <div>
      test
    </div>
  </section>
</div>
`;

exports[`XUIModal should render an automation id when a qaHook is passed in 1`] = `
<div
  aria-hidden={true}
  className="xui-mask xui-unmask"
  data-automationid="xui-modal--mask"
  onClick={null}
  role="presentation"
>
  <section
    className="xui-modal xui-modal-width-medium xui-modal-layout"
    data-automationid="xui-modal"
    role={null}
    style={null}
    tabIndex={-1}
  >
    <header
      className="xui-modal--header xui-modal--heading"
      data-automationid="xui-modal--header"
    >
      <button
        className="xui-button xui-modal--close xui-button-icon xui-button-icon-medium"
        data-automationid="xui-modal--close"
        disabled={false}
        onClick={[Function]}
        tabIndex={0}
        title="Close"
        type="button"
      >
        <div
          className="xui-iconwrapper xui-iconwrapper-medium"
        >
          <svg
            className="xui-icon"
            focusable="false"
            height={12}
            viewBox="0 0 12 12"
            width={12}
          >
            <path
              d="M6 4.5L10.5 0 12 1.5 7.5 6l4.5 4.5-1.5 1.5L6 7.5 1.5 12 0 10.5 4.5 6 0 1.5 1.5 0 6 4.5z"
              role="presentation"
            />
          </svg>
        </div>
      </button>
    </header>
    <div>
      test
    </div>
  </section>
</div>
`;
